process_blocking.go

```diff
@@ -304,19 +304,19 @@ func (p *blockingProcess) Signal(ctx context.Context, sig syscall.Signal) error
 		case res := <-out:
 			return res
 		case <-ctx.Done():
-			return errors.New("context canceled")
+			return errors.Wrap(ctx.Err(), "context canceled")
 		case <-p.complete:
 			// If the process is complete because the operations channel
 			// signaled the process, the signal was successful.
 			if p.Info(ctx).ExitCode == int(sig) {
 				return nil
 			}
-			return errors.New("cannot signal after process is complete")
+			return errors.New("cannot signal a process that has already exited")
 		}
 	case <-ctx.Done():
-		return errors.New("context canceled")
+		return errors.Wrap(ctx.Err(), "context canceled")
```

Comment:
Just return `ctx.Err()`

Reply:
Same as above. I wrapped it with a better message.